ava-labs/core-mobile
# CP-8236 - Improve connection UX where users are clicking "Core" and getting an error

## Description

**Ticket: [CP-8236]** 

Idea was to mock `window.ethereum` object so that we can intercept and handle any calls to it which is initiated by trying to connect to "browser extensions" wallet instead of using WalletConnect. 
We leverage WebView's `injectedJavaScript` property where we can construct such js code which will do aforementioned.
When this is detected we show new dialog `UseWalletConnectModal`

## Screenshots/Videos
[Drive folder](https://drive.google.com/drive/folders/1IfGD74CCAEF50gEMQTGR7OWc4vGyQssq?usp=drive_link) with video on how particular dApp behaves with current changes.
### Observations
Some dApps such `gmx.io` have Core button but it's not correctly implemented, while some such `traderjoexyz` correctly link button to our mobile app.
`multicahin.org` has totally broken dialog.
On `pangolin.exchange` we successfully show dialog but there's no option to connect with WalletConnect.
Majority fall into category where there's no Core button but there's Browser wallet button, witch we successfully handle.

Surprisingly, on core.app things seem to be more complicated than on any other app.
Selecting Core button user is redirected to chrome plug-in store. 
Selecting Core mobile user is presented with another list of options, amongst which is another Core button which this time correctly initiates connection with CoreMobile.

## Testing
- open In-app Browser
- navigate to dApp (Benqi for example)
- select Connect Wallet 
- select Core (extension)
=> Modal dialog should pop up instructing user to select WalletConnect instead

## Checklist

Please check all that apply (if applicable)


[CP-8236]: https://ava-labs.atlassian.net/browse/CP-8236?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
